src/gleam_stdlib.erl

```diff
@@ -536,3 +537,36 @@ base16_decode(String) ->
 
 string_replace(String, Pattern, Replacement) ->
     string:replace(String, Pattern, Replacement, all).
+
+extract_bits(Haystack, Pos, Len) ->
+    <<_:Pos, Needle:Len/bits, _/bits>> = Haystack,
+    Needle.
+
+bit_array_index_of(Haystack, Needle) ->
+    HaystackSize = bit_size(Haystack),
+    NeedleSize = bit_size(Needle),
+    bit_array_find_needle(Haystack, Needle, 0, HaystackSize, NeedleSize).
+
+bit_array_find_needle(_, _, Pos, HaystackSize, NeedleSize) when HaystackSize < NeedleSize + Pos ->
+    -1;
+
+bit_array_find_needle(Haystack, Needle, Pos, HaystackSize, NeedleSize) ->
+    case extract_bits(Haystack, Pos, NeedleSize) of
+        Needle -> Pos;
+        _ -> bit_array_find_needle(Haystack, Needle, Pos + 1, HaystackSize, NeedleSize)
+    end.
+
+bit_array_split_once(Haystack, Needle) ->
```

Comment:
Use the built in split function please 🙏

Reply:
The built in function doesn't seem to allow splitting along non-byte aligned boundaries.

e.g. splitting
<<1, 2, 3, 4, 5>>, with <<3:2>> returns {error, nil}
Current implementation correctly splits it as:
<<1, 2, 0:6>>, <<4, 5>>

Reply:
That's find 👍 Performance is the priority here, and I'm not aware of any binary protocol that will use non-byte aligned data.

Reply:
No problem, i removed the associated tests, and switched to using erlang's built in binary:split